### PR TITLE
Improves the RCD

### PR DESCRIPTION
**Pull request by ike709**
All RCD modes are implemented, RCD ammo exists, and the RCD can no longer build walls on blocked tiles.

There is a bug with @Acruid 's helper method that prevents the RCD from working on a tile adjacent to a wall, which I don't know how to fix. I did, however, fix the helper method's returns being inverted.

### CODE REVIEW

File: Content.Server/GameObjects/Components/Items/RCDComponent.cs [hunk 1]
```diff
@@ -102,50 +105,95 @@ public void Examine(FormattedMessage message, bool inDetailsRange)
...
+                    }
+                    else //Delete what the user targeted
+                    {
+                        //Don't delete mobs or puddles
+                        if (eventArgs.Target == null || eventArgs.Target.TryGetComponent(out PuddleComponent puddleComponent) || eventArgs.Target.TryGetComponent<BodyManagerComponent>(out var bodyManagerComponent))
+                        {
+                            return;
+                        }
+
+                        eventArgs.Target.Delete();
```
Comment (Zumorica) [inline Content.Server/GameObjects/Components/Items/RCDComponent.cs:157]: I don't know, I feel like this should be the opposite: You should only be able to delete entities marked as being deleteable by the RCD

Comment (ike709) [reply Content.Server/GameObjects/Components/Items/RCDComponent.cs:157]: Note to self: RCDDeconstructWhitelistComponent

**Comment by ike709** [comment]
Pretty sure the test failing is not my fault.

File: Resources/Textures/Objects/Tools/rcd.rsi/meta.json [hunk 2]
```diff
@@ -1 +1,63 @@
-{"version": 1, "size": {"x": 32, "y": 32}, "states": [{"name": "inhand-left", "directions": 4, "delays": [[1.0], [1.0], [1.0], [1.0]]}, {"name": "inhand-right", "directions": 4, "delays": [[1.0], [1.0], [1.0], [1.0]]}, {"name": "rcd", "directions": 1, "delays": [[1.0]]}]}
\ No newline at end of file
+{
+    "version": 1,
+    "size": {
```
Comment (Zumorica) [inline Resources/Textures/Objects/Tools/rcd.rsi/meta.json:3]: Add license and copyright to this, please!

**Comment by Zumorica** [comment]
Fix the conflict and I'll merge this right away!

**Comment by RemieRichards** [comment]
Would have preferred `RCDDeconstructible`

**Pull request merged**